File: backend/schemas/app_state.py
from enum import StrEnum
from typing import Annotated

from pydantic import BaseModel, Field

from backend.schemas.suggestion import Suggestion
from backend.schemas.transcript import Transcript


class RunningState(StrEnum):
    IDLE = "idle"
    STARTING = "starting"
    RUNNING = "running"
    STOPPING = "stopping"
    STOPPED = "stopped"


class AppState(BaseModel):
    """The app state."""

    transcripts: Annotated[
        list[Transcript],
        Field(description="The list of transcripts"),
    ]
    running_state: Annotated[
        RunningState,
        Field(description="The running state"),
    ]
    suggestions: Annotated[
        list[Suggestion],
        Field(description="The list of suggestions"),
    ]
    is_backend_live: Annotated[
        bool,
        Field(description="Whether the backend is live"),
    ]

File: ui/types/appState.ts
import { Suggestion } from './suggestion';
import { Transcript } from './transcript';

export enum RunningState {
  IDLE = 'idle',
  STARTING = 'starting',
  RUNNING = 'running',
  STOPPING = 'stopping',
  STOPPED = 'stopped',
}

export interface AppState {
  is_logged_in: boolean | null;
  assistant_state: RunningState;
  transcripts: Transcript[];
  suggestions: Suggestion[];
  is_backend_live: boolean;
  is_gpu_server_live: boolean;
}
